### ISSUE TITLE
Broken UI and bunch of error messages in console

### ISSUE THREAD

**Issue by UgisBrekis** [godotengine/godot#6974]
**Operating system or device - Godot version:**
Windows 10, Godot Alpha 2.2 (October 29th- 30th)

**Issue description** (what happened, and what was expected):
After building latest Godot and running it, the UI seems to be broken. 

The issue appears right away in the Project Manager: vertical slider of the project list is located in the middle of the list container, even though it should be aligned to the side.

Next, scene tree simply doesn't work. The slider is aligned ok, however when I do scrolling the content doesn't scroll, just the scrollbar. When I select item, it is being selected in the 2d editor, however there is no indication in the scene tree. 
Also two buttons overlap in the scene tree panel-> button with the plus icon and with a script icon.

Inspector is broken as well- the same as with the scene tree: there is no indication that something is being selected, some icons seems to be using hi-dpi and some low-dpi icon sets, however the scrolling is working.

Other panels(Node, File System) are simply just messed up, everything is overlapping and it's impossible to make sense of anything there. 

Also, after opening the project, the editor freezes for some time and when it gets back responsive, I noticed there are lots of error messages in the console; here are a couple of them:
failed method: TextureProgress:_update_callback target ID: 1640

> failed method: TextureProgress:_update_callback target ID: 1640
> TOTAL BYTES: 1048572
> NULL count: 8
> CALL : 87206
> CALL _update_callback: 52
> CALL _sort_children: 17
> CALL _update_minimum_size: 18
> ERROR: MessageQueue::push_call: Condition ' (buffer_end+room_needed) >= buffer_size ' is true. returned: ERR_OUT_OF_MEMORY

> failed method: HBoxContainer:_sort_children target ID: 1671
> TOTAL BYTES: 1048572
> NULL count: 8
> CALL : 87215
> CALL _update_callback: 45
> CALL _sort_children: 15
> CALL _update_minimum_size: 18
> ERROR: MessageQueue::push_call: Condition ' (buffer_end+room_needed) >= buffer_size ' is true. returned: ERR_OUT_OF_MEMORY

Basically it's the same error on and on just for the different nodes. 

**Steps to reproduce:**
Simply build the latest Godot alpha v2.2 and run it

**Link to minimal example project** (optional but very welcome):

**Comment by akien-mga** [comment]
I've seen that too, for me the troubles started happening when redocking the Filesystem dock.

For the first thing you mentioned (broken sliders), I've opened #6972 already.

**Comment by UgisBrekis** [comment]
I also now noticed that it takes quite a bit longer for the project to play and debugging isn't working in a way that nothing gets printed in the editor output. I can see printed lines in the console, but output stays empty. 

Regarding your issue with the broken sliders, I think we have to separate issues here- yours is related to drawing while mine is related to the positioning. I tested out to scale them and everything seems to work.

**Comment by akien-mga** [comment]
The slow debugging is due to #6933.

**Comment by volzhs** [comment]
I found it happens with a23d8eafa32153a58e921194c03143d4b8dd037d

**Comment by akien-mga** [comment]
Confirmed that reverting 6ed15e9 fixes the bug for me.

CC @razvanc-r.

**Comment by olsonjeffery** [comment]
I also see the slow startup (but after the slower start everything works fine), along w/ tons of error output to the terminal when godot is ran from the CLI in OSX Sierra, 8321e48a .

A small sample of the output:

````
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
ERROR: set_render_target: Condition ' rt->fbo==0 ' is true.
   At: drivers/gles2/rasterizer_gles2.cpp:4400.
ERROR: _draw_viewport: Condition ' desired_rect.width<=0 || desired_rect.height<=0 ' is true.
   At: servers/visual/visual_server_raster.cpp:7154.
````

**Comment by akien-mga** [comment]
Reverted the problematic change for now.

**Comment by branegames** [comment]
I've been having the same issues with the 3.0.4  when I try to instance additional Control nodes to existing Container node.

Full text from terminal here: https://pastebin.com/wBUP17n4

Does anyone know what causes this issue and how to overcome it?

**Comment by akien-mga** [comment]
Please open a new issue, this one has been closed for 2 years and the error, though similar, is not the same.
